### PR TITLE
Improve diff for env

### PR DESCRIPTION
**Pull request by chesleybrown**
# What

- improves the diff before deployment
- will handle arrays of objects that have `name` properties better by using the `name` as the index
- this means changing the order of things like your ENV values won't generate any differences on deployment as ordering of ENV values in your deployment files does not make any differences

# Example

With current diff:

```
[2016-07-04 17:22:23.730] [INFO] [default] - testing-2-cluster - Differences for activity-deployment: [
  {
    "kind": "E",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      0,
      "env",
      23,
      "name"
    ],
    "lhs": "HOST",
    "rhs": "BIG_MONGO_POOL_SIZE"
  },
  {
    "kind": "E",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      0,
      "env",
      23,
      "value"
    ],
    "lhs": "http://activity.invision.works",
    "rhs": "10"
  },
  {
    "kind": "E",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      0,
      "env",
      24,
      "name"
    ],
    "lhs": "AUTH_PUB_KEY",
    "rhs": "HOST"
  },
  {
    "kind": "E",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      0,
      "env",
      24,
      "value"
    ],
    "lhs": "pubkeyhere",
    "rhs": "http://activity.invision.works"
  },
  {
    "kind": "A",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      0,
      "env"
    ],
    "index": 25,
    "item": {
      "kind": "N",
      "rhs": {
        "name": "AUTH_PUB_KEY",
        "value": "pubkeyhere"
      }
    }
  }
]
```

With new diff changes:

```
[2016-07-04 17:21:52.785] [INFO] [default] - testing-2-cluster - Differences for activity-deployment: [
  {
    "kind": "N",
    "path": [
      "spec",
      "template",
      "spec",
      "containers",
      "activity-con",
      "env",
      "BIG_MONGO_POOL_SIZE"
    ],
    "rhs": {
      "name": "BIG_MONGO_POOL_SIZE",
      "value": "10"
    }
  }
]
```

### CODE REVIEW

File: src/util/manifest-diff.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+const _ = require("lodash");
+const traverse = require("traverse");
+const diff = require("deep-diff");
+
+function traverseAndSort(obj) {
+	traverse(obj).forEach(function(x) {
+		if (_.isArray(x)) {
+			const updated = _.mapKeys(x, function(value, key) {
```
Comment (c1freitas) [inline src/util/manifest-diff.js]: I dont understand this, it looks like it will update everything in the array. Maybe I am missing something..

Comment (chesleybrown) [inline src/util/manifest-diff.js]: Basically any Array it comes across, it looks at all the objects (if any) in that array and if it has the `name` property, then it converts that Array to an Object with "name" being the index for each property.

With that said, you just made me realize an issue. I think it still converts the Array to an Object even in the event there are no `name` properties within the array. I will fix this now.

Comment (chesleybrown) [inline src/util/manifest-diff.js]: Fixed in 4aaeaeb.

Comment (c1freitas) [inline src/util/manifest-diff.js]: Ok, so it seems like it's not actually sorting anything, maybe the name of the function confused me.

Comment (chesleybrown) [inline src/util/manifest-diff.js]: Ah right. I did _originally_ have it sorting the Array but after some testing I changed how it worked but didn't rename the function.

Think I will rename function to `groupArraysByName`? Kind of hard to have a clear function name... lol

Comment (c1freitas) [inline src/util/manifest-diff.js]: Even just a brief comment on what its doing would probably be fine...

Comment (chesleybrown) [inline src/util/manifest-diff.js]: Of course!

Improved in 4483081.

**Comment by c1freitas** [comment]
Looks awesome. 👍

**Pull request merged**